### PR TITLE
Console: Make unsaved Api change work with debug mode

### PR DESCRIPTION
**Pull request by ThibaudAV**
**Issue**
https://github.com/gravitee-io/issues/issues/6768

**Description**

Make unsaved Api change work with debug mode

**Additional context**

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-pokdxoarue.chromatic.com)

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/debug-mode-improvement-2/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

### CODE REVIEW

File: gravitee-apim-console-webui/src/services-ngx/debug-api.service.ts [hunk 1]
```diff
@@ -37,7 +37,11 @@ export class DebugApiService {
     },
   ): Observable<Event> {
     // clean endpoint http proxy
+    // eslint:disable-next-line:@typescript-eslint/ban-ts-comment
+    // @ts-ignore
     if (api.proxy && api.proxy.endpoints) {
+      // eslint:disable-next-line:@typescript-eslint/ban-ts-comment
+      // @ts-ignore
```
Comment (ytvnr) [inline gravitee-apim-console-webui/src/services-ngx/debug-api.service.ts:44]: :o

File: gravitee-apim-console-webui/src/services-ngx/debug-api.service.spec.ts [hunk 2]
```diff
@@ -75,7 +77,7 @@ describe('DebugApiService', () => {
         body: '{"key": "value"}',
       };
 
-      const api = {
+      const api: any = {
```
Comment (ytvnr) [inline gravitee-apim-console-webui/src/services-ngx/debug-api.service.spec.ts:80]: why not an Api type ?

**Review by ytvnr — APPROVED** [review]

File: gravitee-apim-console-webui/src/services-ngx/debug-api.service.ts (hunk 1 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/services-ngx/debug-api.service.ts:44]: 🤷‍♂️  I prefer not to touch this code. and to review together later

File: gravitee-apim-console-webui/src/services-ngx/debug-api.service.spec.ts (hunk 2 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/services-ngx/debug-api.service.spec.ts:80]: test linked to your comment upper. i think is a legacy stuff to remove but 🤷‍♂️ :not-sure: wee can see it in dedicated PR no ?

Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/services-ngx/debug-api.service.spec.ts:80]: i love this emoji 🤷‍♂️

File: gravitee-apim-console-webui/src/services-ngx/debug-api.service.ts (hunk 1 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/services-ngx/debug-api.service.ts:44]: but if you tell me to "delete everything there is no risk". i put all this to 🚮

**Review by gaetanmaisse — APPROVED** [review]

**Pull request merged**